File: Java/algorithms/FizzBuzz.java
// Problem: Print all integers in a given range, but for multiples of 3 print
// "Fizz" and for multiples of 5 print "Buzz". For multiples of 3 and 5 print
// "FizzBuzz".

public class FizzBuzz
{
    public static void printRange(int range)
    {
        for (int i=0; i<=range; i++)
        {
            String output = "";
            if (i % 3 == 0)
                output += "Fizz";
            if (i % 5 == 0)
                output += "Buzz";
            if (output == "")
                output += Integer.toString(i);
            System.out.println(output);
        }
    }

    public static void main(String[] args)
    {
        int range = 100;
        printRange(range);
    }
}

File: Python/TestDrivenDev/FizzBuzz/FizzBuzz.py
# Print values from 1 to 100 - for multiples of 3 print "Fizz", for multiples of
# 5 print "Buzz", for multiples of 3 AND 5 print "FizzBuzz".

def FizzBuzz(val):
    output = ''
    if val % 3 == 0:
        output += 'Fizz'
    if val % 5 == 0:
        output += 'Buzz'
    if not output:
        return str(val)

    return output

def main():
    for i in xrange(100):
        print FizzBuzz(i)

if __name__ == '__main__':
    main()

